open-apparel-registry/open-apparel-registry
# Replace map component

## Overview

This PR replaces the existing Google Map component with a React Leaflet map using a Google Basemap (via GoogleMutant, via some React wrappers). I believe it has feature parity with the Google Map component version on staging, although the cluster marker sizes may be slightly different.

Connects #706 

## Notes

Leaflet.markercluster wants a peer dependency of Leaflet 1.3.something. There's an open PR to fix this by adjusting how it declares its peer dependencies, but the yarn install output here will currently report an incorrect peer dependency. Here's the PR:

https://github.com/Leaflet/Leaflet.markercluster/pull/946

The library seems to work regardless of the peer dependency mismatch, however.

## Testing Instructions

- get this branch, then `./scripts/update` and `./scripts/resetdb` and `./scripts/server`
- use the map and verify that searching for facilities works as before and the clusters work as before
- using `shell_plus`, set two facilities to live at `Point(0, 0)`. I did this just to make them easily searchable:

```
>>> from django.contrib.gis.geos import Point
>>> f = Facility.objects.first()
>>> f.location = Point(0, 0)
>>> f.name = "Hello"
>>> f.save()
>>> f = Facility.objects.last()
>>> f.location = Point(0, 0)
>>> f.name = "Hello world"
>>> f.save()
```

- search for `Hello` in the app and verify that you see two facilities at the same point and that the disambiguation thing works the same as it does on staging

## Checklist
